app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -67,15 +67,18 @@ public class EventsFragment extends Fragment {
         @Override
         public void onMapReady(GoogleMap googleMap) {
             mMap = googleMap;
+            if (ActivityCompat.checkSelfPermission(getActivity(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
This line is WAAAY too long. Consider shortening it, but what might be better is if you make a util method for
this.isPermissionGranted(Manifest.permission.Access_fine_location) &&
 this.isPermissionGranted(Manifest.permission.Access_coarse_location)
 
 This has a couple advantages:
 1. You can reuse the method to save typing/refactoring time in other classes if you put this in a common location. This helps makes it way easier to add another permission check in the future.
 2. You make your line here way shorter and easier to understand.
 3. Most importantly, separating the deltas from the boilerplate. When you move out the uninteresting, repeated boilerplate, you make the slight differences more visible (literally one word on this 1000 character long line), which makes it way easier to understand whats happening on this line and debug it. Right now, you have to really dig to understand the differences in the two checks on this line...in fact, I can't even tell where the closures begin and end. You can see how in my version, especially with proper tabbing, you can IMMEDIATELY see we want to check fine and coarse location. This helps prevent copy-paste bugs.